src/parser/Expression.ts

```diff
@@ -1065,7 +1065,10 @@ export class TemplateStringExpression extends Expression {
                     ...items
                 );
             }
-            plus = items.length > 0 ? ' + ' : '';
```

Comment:
@georgejecook this was the issue. The plus was being set _back_ to an empty string for the next zero length item. And since the expressions were side-by-side, the next quasi was zero length.